File: backend/src/main/java/com/planner/entity/Task.java
package com.planner.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@TableName("task")
public class Task {
    @TableId(type = IdType.AUTO)
    private Long id;
    private Long userId;
    private String title;
    private String description;
    private String status;
    private String priority;
    private Integer plannedMinutes;
    private Integer actualMinutes;
    private LocalDateTime startAt;
    private LocalDateTime finishAt;
    private LocalDate dueAt;
    private LocalDate plannedDate;
    private String tags;
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;
}

File: frontend/src/api/modules/task.ts
import { get, post, put, del } from '../request'

export const taskApi = {
  today: () => get<Task[]>('/tasks/today'),
  list: (params: any) => get<TaskPage>('/tasks', { params }),
  get: (id: number) => get<Task>(`/tasks/${id}`),
  create: (data: any) => post<Task>('/tasks', data),
  update: (id: number, data: any) => put<Task>(`/tasks/${id}`, data),
  updateStatus: (id: number, status: string) => put(`/tasks/${id}/status`, { status }),
  start: (id: number) => put(`/tasks/${id}/start`),
  finish: (id: number) => put(`/tasks/${id}/finish`),
  delete: (id: number) => del(`/tasks/${id}`),
  subTasks: (id: number) => get<SubTask[]>(`/tasks/${id}/subtasks`),
  updateSubTask: (subTaskId: number, done: boolean) => put(`/tasks/subtasks/${subTaskId}`, { done })
}

export interface Task {
  id: number
  title: string
  description?: string
  status: 'todo' | 'doing' | 'done' | 'abandoned'
  priority: 'low' | 'medium' | 'high' | 'urgent'
  plannedMinutes?: number
  actualMinutes?: number
  startAt?: string
  finishAt?: string
  dueAt?: string
  plannedDate?: string
  createdAt: string
  updatedAt: string
}

export interface SubTask {
  id: number
  taskId: number
  title: string
  done: boolean
}

export interface TaskPage {
  records: Task[]
  total: number
  page: number
  size: number
}